cloud/azure/modules/app/main.tf

```diff
@@ -55,107 +55,38 @@ resource "azurerm_subnet" "server_subnet" {
   }
 }
 
-resource "azurerm_subnet" "canary_subnet" {
-  name                 = "canary-subnet"
-  resource_group_name  = data.azurerm_resource_group.rg.name
-  virtual_network_name = azurerm_virtual_network.civiform_vnet.name
-  address_prefixes     = var.canary_subnet_address_prefixes
-
-  delegation {
-    name = "app-service-delegation"
-
-    service_delegation {
-      name    = "Microsoft.Web/serverFarms"
-      actions = ["Microsoft.Network/virtualNetworks/subnets/action"]
-    }
-  }
-}
-
-resource "azurerm_app_service_plan" "plan" {
+resource "azurerm_service_plan" "plan" {
   name                = "${data.azurerm_resource_group.rg.name}-plan"
   location            = data.azurerm_resource_group.rg.location
   resource_group_name = data.azurerm_resource_group.rg.name
-
-  # Define Linux as Host OS
-  kind     = "Linux"
-  reserved = true # Mandatory for Linux plans
-
-  # Choose size
-  sku {
-    tier     = var.app_sku["tier"]
-    size     = var.app_sku["size"]
-    capacity = var.app_sku["capacity"]
-  }
+  os_type             = "Linux"
+  sku_name            = var.app_sku
 }
 
-resource "azurerm_app_service" "civiform_app" {
+resource "azurerm_linux_web_app" "civiform_app" {
   name                = "${var.application_name}-${random_pet.server.id}"
   location            = data.azurerm_resource_group.rg.location
   resource_group_name = data.azurerm_resource_group.rg.name
-  app_service_plan_id = azurerm_app_service_plan.plan.id
-
-  app_settings = local.app_settings
+  service_plan_id     = azurerm_service_plan.plan.id
+  app_settings        = local.app_settings
 
   site_config {
-    linux_fx_version       = "DOCKER|civiform/civiform:${var.image_tag}"
-    always_on              = true
-    vnet_route_all_enabled = true
-  }
-
-  # We will only mount this storage container if SAML authentication is being used
```

Comment:
I think the diff is confusing because I deleted the canary app service entirely, which included the "dynamic storage account" line. You can see that below on line 157 we do preserve the dynamic storage account but it's muddled between what's kept and what's deleted between the two app service definitions.